internal/instrument/instrument.go

```diff
@@ -965,23 +983,60 @@ func ginMiddleware(got *dst.AssignStmt) dst.Stmt {
 	if !ok {
 		return nil
 	}
+	stmt := useMiddleware(iden.Name, "GinMiddleware")
+	wrap(stmt)
+	return stmt
+}
+
+func isEcho(stmt *dst.AssignStmt) bool {
+	rhs := stmt.Rhs[0]
+	f, ok := funcIdent(rhs)
+	return ok && f.Path == "github.com/labstack/echo/v4" && f.Name == "New"
+}
+
+func echoMiddleware(got *dst.AssignStmt) dst.Stmt {
+	iden, ok := got.Lhs[0].(*dst.Ident)
+	if !ok {
+		return nil
+	}
+	stmt := useMiddleware(iden.Name, "EchoMiddleware")
+	wrap(stmt)
+	return stmt
+}
+
+func funcIdent(e dst.Expr) (*dst.Ident, bool) {
+	call, ok := e.(*dst.CallExpr)
+	if !ok {
+		return nil, false
+	}
+	f, ok := call.Fun.(*dst.Ident)
+	if !ok {
+		return nil, false
+	}
+	return f, true
+}
+
+func useMiddleware(pkg, middleware string) *dst.ExprStmt {
 	stmt := &dst.ExprStmt{
 		X: &dst.CallExpr{
 			Fun: &dst.SelectorExpr{
-				X:   &dst.Ident{Name: iden.Name},
+				X:   &dst.Ident{Name: pkg},
 				Sel: &dst.Ident{Name: "Use"},
 			},
 			Args: []dst.Expr{
 				&dst.CallExpr{
 					Fun: &dst.Ident{
-						Name: "GinMiddleware",
+						Name: middleware,
 						Path: "github.com/datadog/orchestrion/instrument",
 					},
 				},
 			},
 		},
 	}
+	return stmt
+}
+
+func wrap(stmt dst.Stmt) {
```

Comment:
Was this moved to a separate function to be reused in other parts of the code other than `useMiddleware` ?

Reply:
@ahmed-mez Yes, that's my intention. As `funcIdent`, it could belong in a separate file for functions that navigate (or manipulate) the AST.